src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { getTodos, getUser } from './api';
+import { TodoStatus } from './enums/TodoStatus';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterParam, setFilterParam] = useState<string>(TodoStatus.All);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      setIsLoading(true);
+      getUser(selectedTodo.userId)
+        .then(setUser)
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedTodo]);
+
+  const getFilteredTodos = useCallback(() => {
```

Comment:
```suggestion
  const filteredTodos = useMemo(() => {
```
in this case better to use useMemo hook. wth use memo no need to call funciton.